### PR TITLE
Add Brave instrumentation for HttpServer

### PR DESCRIPTION
**Pull request by violetagg**
Related to #539

### CODE REVIEW

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java [hunk 1]
```diff
@@ -290,6 +290,21 @@ public final HttpServer httpRequestDecoder(Function<HttpRequestDecoderSpec, Http
...
+	 * See {@link #handle(BiFunction)}.
+	 *
+	 * @param mapHandle A function to decorate the configured I/O handler
+	 * @return a new {@link HttpServer}
+	 */
+	public final HttpServer mapHandle(Function<? super Mono<Void>, ? extends Mono<Void>> mapHandle) {
+		Objects.requireNonNull(mapHandle, "mapHandle");
+		HttpServer dup = duplicate();
+		Function<? super Mono<Void>, ? extends Mono<Void>> current = configuration().mapHandle;
+		dup.configuration().mapHandle = current == null ? mapHandle : current.andThen(mapHandle);
```
Comment (simonbasle) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java:304]: do you foresee a case where one would want to remove mapHandle `Function` (or part(s) of it)? after it has been set?

File: reactor-netty-http/src/main/java/reactor/netty/http/server/Http2StreamBridgeServerHandler.java [hunk 2]
```diff
@@ -40,17 +42,20 @@
 	final ServerCookieEncoder                                     cookieEncoder;
 	final ConnectionObserver                                      listener;
 	final BiFunction<ConnectionInfo, HttpRequest, ConnectionInfo> forwardedHeaderHandler;
+	final Function<? super Mono<Void>, ? extends Mono<Void>> mapHandle;
```
Comment (simonbasle) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/Http2StreamBridgeServerHandler.java:45]: nitpick: breaking the nice alignment here :p

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java [hunk 3]
```diff
@@ -553,22 +560,25 @@ public void handlerAdded(ChannelHandlerContext ctx) {
 		final ConnectionObserver                                      listener;
 		final ServerCookieEncoder                                     cookieEncoder;
 		final ServerCookieDecoder                                     cookieDecoder;
+		final Function<? super Mono<Void>, ? extends Mono<Void>> mapHandle;
```
Comment (simonbasle) [inline reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java:563]: alignment here too

**Review by simonbasle — APPROVED** [review]
only small formatting nitpick, plus a separate comment on "removing" `mapHandle`. From my perspective, looks good.

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java (hunk 3 above)
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java]: fixed

File: reactor-netty-http/src/main/java/reactor/netty/http/server/Http2StreamBridgeServerHandler.java (hunk 2 above)
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/Http2StreamBridgeServerHandler.java]: fixed

**Review by simonbasle — APPROVED** [review]

File: reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java (hunk 1 above)
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java:304]: For the moment no, but if it is needed we can add an API.
I will update the javadoc for this that is adds this function to the existing functions (if any)
wdyt?

Comment (simonbasle) [reply reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java:304]: ah yeah that's probably worth mentioning in the javadoc

**Review by marcingrzejszczak — APPROVED** [review]
At this point LGTM. I'll be able to tell more once I take it out for a spin in Sleuth.

**Review by simonbasle — APPROVED** [review]

**Pull request merged**